cpp/src/arrow/csv/chunker.cc

```diff
@@ -235,10 +235,10 @@ class LexingBoundaryFinder : public BoundaryFinder {
     return Status::OK();
   }
 
-  Status FindNth(util::string_view partial, util::string_view block, uint64_t count,
-                 int64_t* out_pos, uint64_t* num_found) override {
+  Status FindNth(util::string_view partial, util::string_view block, int64_t count,
```

Comment:
Out of curiosity why use signed types for values which should never be negative?

Reply:
That's the convention we use across the Arrow codebase (for example all lengths and offsets are signed, even though they are positive). Given that C++ (and especially some compilers) can be annoying with implicit casts or mixed signed-unsigned operations, it's better not to introduce unsigned variables gratuitously.

Reply:
Should I add asserts or checks that negative values are never passed in then because right now the code does not handle negative values?

Reply:
This is an internal API. We may want to do it at a higher-level, for example in the CSV reader.

Reply:
Looking at the csv options there currently is no validation on any of those values other than asserts. It might make sense to add a options validation to the ReaderMixin as part of another PR.

Mildly related, I also noticed I probably should have made `skip_rows_after_names` an `int64_t` to match the type of the variable in the chunker. Do you want me to make that change quickly or do you want to do it?

Reply:
We can keep it a `int32_t` for consistency with `skip_rows`. There is zero use case for skipping more than 2**31 rows :-)

Reply:
You're right about options validation. Do you want to open a separate JIRA for it?